sdcm/mgmt/cli.py

```diff
@@ -202,7 +246,12 @@ def progress(self) -> str:
     def detailed_progress(self):
         if self.status in [TaskStatus.NEW, TaskStatus.STARTING]:
             return " 0%"
-        cmd = "task progress {} -c {}".format(self.id, self.cluster_id)
+
+        if self.sctool.parsed_client_version >= new_command_structure_minimum_version:
```

Comment:
seems like a code duplication. Please add a function that checks the version and returns the command and call it here.